bigquery/google/cloud/bigquery/client.py

```diff
@@ -1929,6 +1935,61 @@ def list_rows(
         )
         return row_iterator
 
+    def _schema_from_json_file_object(self, file):
+        """Helper function for schema_from_json that takes a
+           file object that describes a table schema.
+
+           Returns:
+                List of schema field objects.
+        """
+        schema_field_list = list()
+        json_data = json.load(file)
+
+        for field in json_data:
+            schema_field = SchemaField.from_api_repr(field)
+            schema_field_list.append(schema_field)
+
+        return schema_field_list
+
+    def schema_from_json(self, file_or_path):
+        """Takes a file object or file path that contains json that describes
+            a table schema.
+
+            Returns:
+                List of schema field objects.
+        """
+        if isinstance(file_or_path, io.IOBase):
+            return self._schema_from_json_file_object(file_or_path)
+        else:
+            try:
+                with open(file_or_path) as file:
+                    return self._schema_from_json_file_object(file)
+            except OSError:
+                raise ValueError(_NEED_JSON_FILE_ARGUMENT)
+
+    def schema_to_json(self, schema_list, destination):
+        """Takes a list of schema field objects.
+
+            Serializes the list of schema field objects as json to a file.
+
+            Destination is a file path or a file object.
+        """
+        file_obj = None
+        json_schema_list = list()
+
+        for field in schema_list:
+            schema_field = field.to_api_repr()
+            json_schema_list.append(schema_field)
+
+        if isinstance(destination, io.IOBase):
+            destination.write(json.dumps(json_schema_list, indent=2, sort_keys=True))
```

Comment:
Use [`json.dump(schema_list, destination)`](https://docs.python.org/3/library/json.html#json.dump) instead. Then you can use `BytesIO` in the tests, too.

I'd prefer if `json.dump` wasn't repeated (here we do want to be DRY). Replace this with `file_obj = destination` and remove the `else`.

Reply:
Refactored into a helper function because I couldn't get it to work with the context manager below just by removing the else.